tests/e2e/csi_cns_telemetry.go

```diff
@@ -0,0 +1,278 @@
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ginkgo.By("Invoking cluster-distribution for volume basic test")
+		var storageclasspvc1 *storagev1.StorageClass
+		var pvclaim1 *v1.PersistentVolumeClaim
+		var err error
+
+		scParameters[scParamDatastoreURL] = datastoreURL
+		storageclasspvc1, pvclaim1, err = createPVCAndStorageClass(client, namespace, nil, scParameters, diskSize, nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		defer func() {
+			err = client.StorageV1().StorageClasses().Delete(ctx, storageclasspvc1.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Expect claim to provision volume successfully")
+		err = fpv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, pvclaim1.Namespace, pvclaim1.Name, framework.Poll, time.Minute)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to provision volume")
+
+		persistentvolumes1, err := fpv.WaitForPVClaimBoundPhase(client, []*v1.PersistentVolumeClaim{pvclaim1}, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		volHandle1 := persistentvolumes1[0].Spec.CSI.VolumeHandle
+		gomega.Expect(volHandle1).NotTo(gomega.BeEmpty())
+
+		defer func() {
+			err = fpv.DeletePersistentVolumeClaim(client, pvclaim1.Name, pvclaim1.Namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(volHandle1)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By(fmt.Sprintf("Invoking QueryCNSVolumeWithResult with VolumeID: %s", volHandle1))
+		queryResult1, err := e2eVSphere.queryCNSVolumeWithResult(volHandle1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(len(queryResult1.Volumes) > 0)
+
+		framework.Logf("Cluster-distribution value on CNS is %s \n", queryResult1.Volumes[0].Metadata.ContainerCluster.ClusterDistribution)
+
+		if queryResult1.Volumes[0].Metadata.ContainerCluster.ClusterDistribution != "CSI-Vanilla" {
+			err = fmt.Errorf("Wrong cluster-distribution name present on CNS")
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		csivsphereConf := "csi-vsphere.conf"
+		// Fetch existing vsphere-config-secret
+		currentSecret, getSecretErr := client.CoreV1().Secrets(kubeSystemNamespace).Get(ctx, configSecret, metav1.GetOptions{})
+		gomega.Expect(getSecretErr).NotTo(gomega.HaveOccurred())
+		framework.Logf("Secret Name is %s\n", currentSecret.Name)
+
+		// Read and map the secret values to cfg
+		originalConf := string(currentSecret.Data[csivsphereConf])
+		cfg, readerr := readConfigFromSecretString(originalConf)
+		gomega.Expect(readerr).NotTo(gomega.HaveOccurred())
+
+		framework.Logf("Check Cluster-distribution value before running the test %s\n", cfg.Global.ClusterDistribution)
+
+		// Check if the cluster-distribution value is CSI-Vanilla or not, if not reset
+		if cfg.Global.ClusterDistribution != "CSI-\\tVanilla-Test" {
```

Comment:
add cleanup for this block in AfterEach() in case test case fails without cleaning up